substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/code/CompilationResultFrameTree.java

```diff
@@ -278,6 +284,48 @@ public String toString() {
             return String.format("-%s, span %d, bci %d, method %s", getPosStr(), getSpan(), frame.getBCI(), frame.getMethod().format("%h.%n(%p)"));
         }
 
+        public String getLocalsStr() {
+            String localsInfo = "";
+            if (frame instanceof BytecodeFrame) {
+                BytecodeFrame bcf = (BytecodeFrame) frame;
+                Local[] locals = getLocalsBySlot(frame.getMethod());
+                StringBuilder sb = new StringBuilder();
+                if (locals == null) {
+                    return localsInfo;
+                }
+                int combinedLength = Integer.min(locals.length, bcf.numLocals);
+                for (int i = 0; i < combinedLength; i++) {
+                    JavaValue value = bcf.getLocalValue(i);
+                    if (i > 0) {
+                        sb.append(", ");
+                    }
+                    sb.append("li(");
+                    Local local = locals != null ? locals[i] : null;
+                    if (local != null) {
+                        sb.append(local.getName());
+                        sb.append("=");
+                    }
+                    sb.append(value);
+                    sb.append(")");
+                }
+                localsInfo = sb.toString();
+            }
+            return localsInfo;
+        }
+
+        private Local[] getLocalsBySlot(ResolvedJavaMethod method) {
+            LocalVariableTable lvt = method.getLocalVariableTable();
+            Local[] nonEmptySortedLocals = null;
+            if (lvt != null) {
+                Local[] locals = lvt.getLocals();
+                if (locals != null && locals.length > 0) {
+                    nonEmptySortedLocals = Arrays.copyOf(locals, locals.length);
+                    Arrays.sort(nonEmptySortedLocals, (Local l1, Local l2) -> l1.getSlot() - l2.getSlot());
```

Comment:
`Arrays.sort(nonEmptySortedLocals, Comparator.comparingInt(Local::getSlot));` is clearly more elegant.